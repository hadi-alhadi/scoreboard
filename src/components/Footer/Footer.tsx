import packageJson from "../../../package.json";
import { Box, Container, Link, Typography } from "@mui/material";

const Copyright = () => {
  const currentYear = new Date().getFullYear();
  return (
    <Typography
      variant="body2"
      color="common.white"
      align="center"
      data-testid="copyright"
    >
      {`Copyright Â© ${packageJson.author} ${currentYear}. All rights reserved.`}
    </Typography>
  );
};

const ProfileLink = (props: { url: string; profileName: string }) => (
  <Link
    href={props.url}
    underline="none"
    color={"inherit"}
    sx={{ mx: 2, my: 1 }}
    target="_blank"
  >
    {props.profileName}
  </Link>
);

const Footer = () => {
  return (
    <Box
      component="footer"
      sx={{ bgcolor: "#550065", py: 6, position: "sticky", top: "100%" }}
      color={"white"}
      data-testid="footer"
    >
      <Container maxWidth="lg">
        <Container sx={{ display: "flex", justifyContent: "center" }}>
          <ProfileLink
            url={"https://linkedin.com/in/hadi-al-hadi"}
            profileName={"LinkedIn"}
          />
          <ProfileLink
            url={"https://github.com/hadi-alhadi"}
            profileName={"GitHub"}
          />
        </Container>

        <Copyright />
      </Container>
    </Box>
  );
};

export default Footer;
