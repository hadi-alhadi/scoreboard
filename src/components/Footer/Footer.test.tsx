import Footer from "./Footer";
import { render, screen } from "@testing-library/react";
import { describe, expect, it, beforeEach } from "vitest";
import packageJson from "../../../package.json";

describe("footer", () => {
  beforeEach(() => {
    render(<Footer />);
  });

  it("should render", () => {
    expect(screen.getByTestId("footer")).toMatchSnapshot();
  });

  it("should render the ProfileLink component", async () => {
    expect(screen.getByRole("link", { name: "LinkedIn" })).toHaveAttribute(
      "href",
      "https://linkedin.com/in/hadi-al-hadi",
    );
    expect(screen.getByRole("link", { name: "GitHub" })).toHaveAttribute(
      "href",
      "https://github.com/hadi-alhadi",
    );
  });

  it("should render the copyright", () => {
    const currentYear = new Date().getFullYear();
    expect(screen.getByTestId("copyright")).toHaveTextContent(
      `Â© ${packageJson.author} ${currentYear}. All rights reserved.`,
    );
  });
});
