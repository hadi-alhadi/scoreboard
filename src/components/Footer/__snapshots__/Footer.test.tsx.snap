// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`footer > should render 1`] = `
<footer
  class="sc-HzRMQ SLHHK MuiBox-root"
  data-testid="footer"
>
  <div
    class="MuiContainer-root-Bmntt bMzjnc MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiContainer-root-Bmntt bUDjfM MuiContainer-root MuiContainer-maxWidthLg"
    >
      <a
        class="MuiTypography-root-ieDiFq czpiQF MuiTypography-root MuiTypography-inherit MuiLink-root-bAfifx iTvSWk MuiLink-root MuiLink-underlineNone"
        href="https://linkedin.com/in/hadi-al-hadi"
        target="_blank"
      >
        LinkedIn
      </a>
      <a
        class="MuiTypography-root-ieDiFq czpiQF MuiTypography-root MuiTypography-inherit MuiLink-root-bAfifx iTvSWk MuiLink-root MuiLink-underlineNone"
        href="https://github.com/hadi-alhadi"
        target="_blank"
      >
        GitHub
      </a>
    </div>
    <p
      class="MuiTypography-root-ieDiFq jcuooA MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter"
      data-testid="copyright"
    >
      Copyright Â© Hadi Alhadi 2023. All rights reserved.
    </p>
  </div>
</footer>
`;
